### ISSUE TITLE
Graph Editor: Crash when opening subgraph with: `constant` node connected to `input` node

### ISSUE THREAD

**Issue by marwie** [AcademySoftwareFoundation/MaterialX#1456]
Opening the nodegraph `needle_shadergraph` crashes the editor

It started happening after adding the `density` input element connected to the `constant`. They are not used in the graph right now but the crash should most likely not happen in any case :)

```xml
<?xml version="1.0"?>
<materialx name="" version="1.38" >
	<nodegraph name="needle_shadergraph" >
		<constant name="color" type="color4" >
			<input name="value" type="color4" value="0.02800465, 0.0, 1.0, 0.0" />
		</constant>
		<texcoord name="uv" type="vector3" >
			<input name="index" type="integer" value="0" />
		</texcoord>
		<!-- MyMultiplier -->
		<input name="_MyMultiplier" type="float" value="36.2" />
		<dot name="const__MyMultiplier" type="float" >
			<input name="in" type="float" interfacename="_MyMultiplier" />
		</dot>
		<constant name="Float" type="float" >
			<input name="value" type="float" nodename="const__MyMultiplier" />
		</constant>
		<convert name="convert_to_vector3" type="vector3" >
			<input name="in" type="float" nodename="Float" />
		</convert>
		<multiply name="mult" type="vector3" >
			<input name="in1" type="vector3" nodename="uv" />
			<input name="in2" type="vector3" nodename="convert_to_vector3" />
		</multiply>
		<constant name="Float_1" type="float" >
			<input name="value" type="float" value="5.0" />
		</constant>
		<input name="density" type="float" nodename="Float_1" />
		<convert name="convert_to_vector2" type="vector2" >
			<input name="in" type="vector3" nodename="mult" />
		</convert>
		<worleynoise2d name="worley" type="float" >
			<input name="texcoord" type="vector2" nodename="convert_to_vector2" />
			<dot name="" type="float" >
				<input name="in" value="1.0" />
			</dot>
		</worleynoise2d>
		<convert name="convert_to_color4" type="color4" >
			<input name="in" type="vector4" value="1.57, 2.6, 1.0, 1.0" />
		</convert>
		<convert name="convert_to_color4_1" type="color4" >
			<input name="in" type="float" nodename="worley" />
		</convert>
		<mix name="mix" type="color4" >
			<input name="fg" type="color4" nodename="color" />
			<input name="bg" type="color4" nodename="convert_to_color4" />
			<input name="mix" type="color4" nodename="convert_to_color4_1" />
		</mix>
		<convert name="convert_to_color3" type="color3" >
			<input name="in" type="color4" nodename="mix" />
		</convert>
		<output name="out" type="color3" nodename="convert_to_color3" />
	</nodegraph>
	<standard_surface name="needle_standard_surface" type="surfaceshader" >
		<!-- BaseColor -->
		<input name="base_color" type="color3" nodegraph="needle_shadergraph" output="out" />
		<!-- Smoothness -->
		<input name="specular_roughness" type="float" value="0.2" />
		<!-- Metallic -->
		<input name="metalness" type="float" value="0.3" />
	</standard_surface>
	<surfacematerial name="Default" type="material" >
		<input name="surfaceshader" type="surfaceshader" nodename="needle_standard_surface" />
	</surfacematerial>
</materialx>
```

**Comment by marwie** [comment]
Here is a smaller version that also already crashes

```xml
<?xml version="1.0"?>
<materialx name="" version="1.38" >
	<nodegraph name="needle_shadergraph" >
		<constant name="Float_1" type="float" >
			<input name="value" type="float" value="5.0" />
		</constant>
		<input name="density" type="float" nodename="Float_1" />
	</nodegraph>
	<standard_surface name="needle_standard_surface" type="surfaceshader" >
		<!-- Smoothness -->
		<input name="specular_roughness" type="float" value="0.2" />
		<!-- Metallic -->
		<input name="metalness" type="float" value="0.3" />
	</standard_surface>
	<surfacematerial name="Default" type="material" >
		<input name="surfaceshader" type="surfaceshader" nodename="needle_standard_surface" />
	</surfacematerial>
</materialx>

```

**Comment by jstone-lucasfilm** [comment]
Thanks for this detailed report, @marwie, and I've proposed a fix for the issue in #1466.

**Comment by jstone-lucasfilm** [comment]
The proposed fix is now merged, and let us know if this addresses the issue for you.
